sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala

```diff
@@ -53,7 +53,11 @@ class PathFilterIgnoreNonData(stagingDir: String) extends PathFilter with Serial
 object CommandUtils extends Logging {
 
   /** Change statistics after changing data by commands. */
-  def updateTableStats(sparkSession: SparkSession, table: CatalogTable): Unit = {
+  def updateTableStats(
+      sparkSession: SparkSession,
+      table: CatalogTable,
+      partitionSpec: Map[String, Option[String]] = Map.empty,
+      withAutoPartitionStats: Boolean = true): Unit = {
```

Comment:
maybe make this default to false so the existing calls `updateTableStats(session, table)` can still maintain the existing behavior.

Reply:
Changed to `isDropPartition: Boolean = false` since the only use case is the optimization for dropping partitions